hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -896,8 +899,10 @@ private AbfsRestOperation conditionalCreateOverwriteFile(final String relativePa
 
         try {
           // overwrite only if eTag matches with the file properties fetched before.
-          if (getPrefixMode() == PrefixMode.BLOB) {
-            boolean isNormalBlob = !checkIsBlobOrMarker(metadata);
+          boolean isNormalBlob = !checkIsBlobOrMarker(metadata);
+          if (!isNormalBlob && !abfsConfiguration.shouldMkdirFallbackToDfs() && getPrefixMode() == PrefixMode.BLOB) {
+            op = createPathBlob(relativePath, isNormalBlob, true, metadata, eTag, tracingContext);
+          } else if (!abfsConfiguration.shouldIngressFallbackToDfs() && getPrefixMode() == PrefixMode.BLOB) {
```

Comment:
Order of check will be good to have PrefixMode first and then fallbackToDFS (as latter applies only for Blob prefix mode)

Reply:
same for createDirectory